debezium-core/src/test/java/io/debezium/transforms/HeaderToValueTest.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.transforms;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.header.Header;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.transforms.util.Requirements;
+import org.junit.Test;
+
+import io.debezium.data.Envelope;
+
+public class HeaderToValueTest {
```

Comment:
Please also add fail test scenarios for cases

* empty value or space is passed
  * In the middle of the string
  * At the beggining or end of the string
  * In the middle of nested address
* field names not separated with comma only but with comma and space

Consider in which scenarios it makes sense to apply trim to strings

Reply:
`At the beggining or end of the string` are already managed by `getList()` of `Configuration` class. Thrown a `ConfigException` in all other cases.